src/gtc/common.py

```diff
@@ -572,6 +572,46 @@ def apply(cls, node: Node, *, symtable: Dict[str, Any]) -> List[str]:
     return root_validator(allow_reuse=True, skip_on_failure=True)(_impl)
 
 
+def validate_lvalue_dims() -> RootValidatorType:
+    """Validate lvalue dimensions using the root node symbol table."""
+
+    class _LvalueDimsValidator(NodeVisitor):
+        def visit_Node(self, node: Node, symtable: Dict[str, Any], **kwargs: Any) -> None:
+            if hasattr(node, "loop_order"):
```

Comment:
Is the mere existence of a `loop_order` attribute a good way to check this? The `VerticalLoop` nodes are different for the different IRs, i.e. have no common base class, but having this check in common suggests that they should. Then an `isinstance` check would be useful. In general I believe duck typing is not the right approach for IR nodes, as IRs are usually closed and type checks increase readability of what is expected.

Reply:
I agree, we really want to ensure that any attribute of type `common.LoopOrder` is passed on, regardless of it's name.

Reply:
The improved version relies purely on the presence of a `LoopOrder` in the tree above the `AssignStmt` and makes no other assumptions about the concrete IR anymore

Reply:
Actually, the same problem applies to the `dimensions` field, so I introduced a new `DimensionFlags` node to make it independent of attribute naming.

Reply:
> Is there a reason for not having tests starting from the frontend yet (e.g. storages not ready)? Rest looks good, only small things.

We have tests for every step along the way and we didn't want to add to the integration test burden unneccessarily.